src/util/InMemoryLogHandler.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright 2017 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "util/InMemoryLogHandler.h"
+#include <sstream>
+
+namespace stellar
+{
+
+InMemoryLogHandler::InMemoryLogHandler(LogBuffer logBuffer)
+    : pushLevel(el::Level::Fatal), buffer(logBuffer)
+{
+    logger = el::Loggers::getLogger(Logging::inMemoryLoggerName);
+    el::Configurations* config = logger->configurations();
+    config->set(el::Level::Global, el::ConfigurationType::Enabled, "true");
+    config->setGlobally(el::ConfigurationType::ToStandardOutput, "false");
+    config->setGlobally(el::ConfigurationType::ToFile, "false");
+    logger->configure(*config);
+}
+
+InMemoryLogHandler::InMemoryLogHandler() : InMemoryLogHandler(LogBuffer(100))
+{
+}
+
+void
+InMemoryLogHandler::handle(const el::LogDispatchData* handlePtr)
+{
+    std::string logLine = buildLogLine(*handlePtr);
+    buffer.push(logLine);
+    if (checkForPush(*handlePtr))
+    {
+        pushLogs();
+    }
+}
+
+std::string
+InMemoryLogHandler::buildLogLine(const el::LogDispatchData& data) const
+{
+    return data.logMessage()->logger()->logBuilder()->build(
+        data.logMessage(),
+        data.dispatchAction() == el::base::DispatchAction::NormalLog);
+}
+
+void
+InMemoryLogHandler::pushLogs()
+{
+    bool notEmpty = !buffer.empty();
+    if (notEmpty)
+    {
+        std::stringstream logStream;
+        logStream << "<in-memory-logger>" << std::endl;
+        std::string logLine = logStream.str();
+        printToLog(logLine);
+    }
+
+    while (!buffer.empty())
+    {
+        auto logLine = buffer.pop();
+        auto newMsg = el::LogMessage(el::Level::Trace, "", 0, "", 1, logger);
+        auto data =
+            el::LogDispatchData(&newMsg, el::base::DispatchAction::NormalLog);
+        dispatchCallback.dispatch(logLine, &data);
+        logLine = "";
+    }
+
+    if (notEmpty)
+    {
+        std::stringstream logStream;
+        logStream << "</in-memory-logger>" << std::endl;
+        std::string logLine = logStream.str();
+        printToLog(logLine);
+    }
+}
+
+void
+InMemoryLogHandler::printToLog(const std::string& message)
+{
+    auto newMsg = el::LogMessage(el::Level::Trace, "", 0, "", 1, logger);
+    auto data =
+        el::LogDispatchData(&newMsg, el::base::DispatchAction::NormalLog);
+    dispatchCallback.dispatch(message, &data);
+}
+
+bool
+InMemoryLogHandler::checkForPush(const el::LogDispatchData& data) const
+{
+    if (data.logMessage()->level() == pushLevel)
```

Comment:
trivial: I think we should use >= or <= here (don't know how the values are set up)

if I set up pushLevel to ERROR, I would also expect FATAL logs to add in-memory logs to files